msal/src/main/java/com/microsoft/identity/client/SingleAccountPublicClientApplication.java

```diff
@@ -95,4 +100,114 @@ public void onError(Exception exception) {
                     }
                 });
     }
+
+    private MsalOAuth2TokenCache getTokenCache(@NonNull final Context appContext,
+                                               @NonNull final IAccountCredentialCache accountCredentialCache) {
+        return new MsalOAuth2TokenCache(
+                appContext,
+                accountCredentialCache,
+                new MicrosoftStsAccountCredentialAdapter());
+    }
+
+    private IAccountCredentialCache getCurrentAccountCredentialCache(final Context appContext){
+        return new SharedPreferencesAccountCredentialCache(
+                new CacheKeyValueDelegate(),
+                new SharedPreferencesFileManager(
+                        appContext,
+                        SINGLE_ACCOUNT_CREDENTIAL_SHARED_PREFERENCES,
+                        new StorageHelper(appContext)
+                )
+        );
+    }
+
+    /**
+     * Get current account that is persisted in shared preference.
+     *
+     * @param appContext
+     * @return a persisted MultiTenantAccount. This could be null.
+     * */
+    @Nullable
+    private MultiTenantAccount getPersistedCurrentAccount(@NonNull final Context appContext) {
+        final String methodName = ":getPersistedCurrentAccount";
+        final IAccountCredentialCache accountCredentialCache = getCurrentAccountCredentialCache(appContext);
```

Comment:
`IAccountCredentialCache` should really be _internal_ to the cache -- `PublicClientApplication` should call methods on `MsalOauth2TokenCache` to serve its needs